Pandas test failing on Pandas 0.18

Found [on Travis here](https://travis-ci.org/HypothesisWorks/hypothesis-python/jobs/284565714); was blocking but otherwise unrelated to #921.

<details>

```
=================================== FAILURES ===================================
_______________________ test_generate_arbitrary_indices ________________________
[gw0] linux -- Python 3.6.1 /home/travis/.runtimes/.tox/pandas18/bin/python
tests/pandas/test_indexes.py:66: in test_generate_arbitrary_indices
    def test_generate_arbitrary_indices(data):
../../../.runtimes/.tox/pandas18/lib/python3.6/site-packages/hypothesis/executors.py:58: in default_new_style_executor
    return function(data)
../../../.runtimes/.tox/pandas18/lib/python3.6/site-packages/hypothesis/core.py:136: in run
    return test(*args, **kwargs)
tests/pandas/test_indexes.py:66: in test_generate_arbitrary_indices
    def test_generate_arbitrary_indices(data):
../../../.runtimes/.tox/pandas18/lib/python3.6/site-packages/hypothesis/core.py:590: in timed_test
    result = test(*args, **kwargs)
tests/pandas/test_indexes.py:100: in test_generate_arbitrary_indices
    assert index.dtype == inferred_dtype
E   AssertionError: assert dtype('O') == dtype('<M8[ns]')
E    +  where dtype('O') = Index([], dtype='object').dtype
---------------------------------- Hypothesis ----------------------------------
Falsifying example: test_generate_arbitrary_indices(data=data(...))
Draw 1 (min_size): 0
Draw 2 (max_size): 0
Draw 3 (unique): False
Draw 4 (dtype): dtype('<M8[ns]')
Draw 5 (pass_elements): True
Draw 6: numpy.datetime64('1970-01-01T00:00:00.000000000')
Draw 7: Index([], dtype='object')
You can add @seed(302934307671667531413257853548643485645) to this test or run pytest with --hypothesis-seed=302934307671667531413257853548643485645 to reproduce this failure.
===================== 1 failed, 79 passed in 98.42 seconds =====================
```

</details>

--- a/tests/pandas/test_indexes.py
+++ b/tests/pandas/test_indexes.py
@@ -97,7 +97,10 @@ def test_generate_arbitrary_indices(data):
         else:
             raise
     if dtype is None:
-        assert index.dtype == inferred_dtype
+        if pandas.__version__ >= '0.19':
+            assert index.dtype == inferred_dtype
+        else:
+            assert inferred_dtype in (index.dtype, np.dtype(object))
     else:
         assert index.dtype == converted_dtype
 

